File: components/workout_data_structs.py

class Workout:
    def __init__(self, exercises=None, date=None, name=None):
        self.exercises = exercises
        self.date = date
        self.name = name

class Exercise:
    def __init__(
            self, 
            name=None, 
            sets=None,
            notes=None
        ):
        self.name = name
        self.sets = sets
        self.notes = notes

class Set:
    def __init__(
            self, 
            reps=None, 
            weight=None, 
            drop_reps=None, 
            drop_weight=None,
            time=None
        ):
        self.reps = reps
        self.weight = weight
        self.drop_reps = drop_reps
        self.drop_weights = drop_weight
        self.time = time

File: components/workout_data_structs.cpp
#include "workout_data_structs.h"
#include "common_types.h"

using cType::RepVal;

Set::Set(
    RepVal reps, 
    RepVal drop_reps,
    double weight, 
    double drop_weight,
    std::string time):
        reps(reps),
        drop_reps(drop_reps),
        weight(weight),
        drop_weight(drop_weight),
        time(time) {} 

// Exercise::Exercise(
//     std::string name,
//     std::vector<Set> sets,
//     std::string notes):
//     name(name),
//     sets(sets),
//     notes(notes) {}